### PR TITLE
Adding mask support

### PR DESCRIPTION
**Pull request by krispypen**
* mask support
* added some test "golden" test files
* also fixed that parentstyles where not used with the "use" element

### CODE REVIEW

**Comment by krispypen** [comment]
@dnfield can you first merge #200 to fix the travis build?

File: lib/src/vector_drawable.dart [hunk 1]
```diff
@@ -1153,6 +1163,15 @@ class DrawableShape implements DrawableStyleable {
     } else {
       innerDraw();
     }
+    if (style.mask != null) {
+      final Paint p = Paint();
+      p.blendMode = BlendMode.dstIn;
+      p.colorFilter = ColorFilter.matrix(<double>[0,0,0,0,0, 0,0,0,0,0, 0,0,0,0,0, 0.21,0.71,0.08,0,0]); //convert grayscale values to transparency
```
Comment (dnfield) [inline lib/src/vector_drawable.dart]: This seems right, but could you provide a link to the spec for this in a comment?

File: lib/src/vector_drawable.dart [hunk 2]
```diff
@@ -1143,6 +1150,9 @@ class DrawableShape implements DrawableStyleable {
       }
     };
 
+    if (style.mask != null) {
+      canvas.saveLayer(bounds, Paint());
```
Comment (dnfield) [inline lib/src/vector_drawable.dart]: Don't pass in the bounds here.  SaveLayer will just use the current clip anyway, and the bounds may not actually be what we want if there are transforms involved.

File: lib/src/vector_drawable.dart [hunk 3]
```diff
@@ -1153,6 +1163,15 @@ class DrawableShape implements DrawableStyleable {
     } else {
       innerDraw();
     }
+    if (style.mask != null) {
+      final Paint p = Paint();
+      p.blendMode = BlendMode.dstIn;
+      p.colorFilter = ColorFilter.matrix(<double>[0,0,0,0,0, 0,0,0,0,0, 0,0,0,0,0, 0.21,0.71,0.08,0,0]); //convert grayscale values to transparency
+      canvas.saveLayer(bounds, p);
```
Comment (dnfield) [inline lib/src/vector_drawable.dart]: Ditto on not passing in the bounds here.

File: lib/src/vector_drawable.dart (hunk 2 above)
Comment (dnfield) [reply lib/src/vector_drawable.dart]: Also, do we really need both of these?  Is this one required?

Comment (krispypen) [reply lib/src/vector_drawable.dart]: You mean both invocations of saveLayer? The first one is needed to let te mask only apply on the current shape rendered in the "innerDraw" method, and the second one is needed to convert the mask to transparency and apply it to the layer

**Comment by dnfield** [comment]
Thanks!

**Pull request merged**